front/src/modules/simulationResult/components/SpaceTimeChartWrapper/SpaceTimeChartWrapper.tsx

```diff
@@ -230,70 +254,93 @@ const SpaceTimeChartWrapper = ({
-                  path,
-                  !!draggingState,
-                  subCategories,
-                  timetableItemsWithDetails,
-                  selectedTrainId
-                );
-                return {
-                  ...zone,
-                  color: pathStyle.color,
-                  size: pathStyle.level === 1 ? 2 : undefined,
-                };
-              })}
-              selectedTrainId={selectedTrainId}
-              onClose={() => onCloseOccupancyLayer?.(waypointId)}
-              topPadding={BASE_WAYPOINT_HEIGHT}
-            />
-          ),
-          manchetteNode: (
-            <TrackOccupancyManchette tracks={tracks || []}>
-              <div className="waypoint-wrapper flex justify-start">
-                <WaypointComponent
-                  waypoint={{
-                    id: waypointId,
-                    name: (
-                      <div className="d-flex flex-row align-items-center">
-                        {operationalPointName || operationalPointId}
-                        {loading && (
-                          <Spinner className="ml-2 small" spinnerClassName="spinner-border-sm" />
-                        )}
-                      </div>
-                    ),
-                    position: operationalPointPosition,
-                    onClick: handleWaypointClick,
-                  }}
-                  waypointRef={activeWaypointRef}
-                  isActive={false}
-                  isMenuActive={false}
-                />
-              </div>
-            </TrackOccupancyManchette>
-          ),
-        })
-      ) || []
-    );
-  }, [
-    trackOccupancyDiagramsData,
-    activeWaypointId,
-    timetableItemsWithDetails,
-    selectedTrainId,
-    hoveredItem,
-  ]);
+        }) => {
+          const baseZones = zones ?? [];
+          const zonesCountByPacedTrainId = countZonesByPacedTrainId(baseZones);
+
+          const styledZones = baseZones.map((zone) => {
+            const path = pathsById[zone.trainId];
+
+            const isHovered = hoveredPathId === zone.trainId;
+            const isSelected = selectedTrainId === zone.trainId;
+
+            let totalOccurrencesOnTrack = 0;
+            if (isOccurrenceId(zone.trainId)) {
+              const pacedTrainId = extractPacedTrainIdFromOccurrenceId(zone.trainId);
+              totalOccurrencesOnTrack =
+                zonesCountByPacedTrainId.get(pacedTrainId)?.get(zone.trackId) ?? 0;
+            }
+            if (!path) return zone;
+
+            const pathStyle = getPathStyle(
+              hoveredItem,
+              path,
+              !!draggingState,
+              subCategories,
+              timetableItemsWithDetails,
+              selectedTrainId
+            );
+            return {
+              ...zone,
+              label:
+                isHovered && totalOccurrencesOnTrack > 1
+                  ? `+${String(Math.max(0, totalOccurrencesOnTrack - 1))}`
```

Comment:
You don't need the String operator here
```suggestion
                  ? `+${Math.max(0, totalOccurrencesOnTrack - 1)}`
```